docs/md/logging.md

```diff
@@ -0,0 +1,68 @@
+# WFL Logging
+
+## Usage
+We use `clojure.tools.logging`.
+
+Require it like any other dependency:
+```clojure
+(ns "..."
+  (:require
+    ...
+    [clojure.tools.logging :as log]
+    ...))
+
+(log/info "Hello!")
+```
+
+Full documentation is available [here](http://clojure.github.io/tools.logging/#clojure.tools.logging),
+but the primary functions we use are these:
+
+- `log/error` and `log/errorf` for failures and uncaught exceptions
+- `log/warn` and `log/warnf` for issues that could be recovered from
+- `log/info` and `log/infof` for recording normal interaction
+- `log/debug` and `log/debugf` for more granular output that might be 
+added during debugging
+
+The first function of each level accepts any number of arguments and 
+concatenates them with spaces, while the second variant uses format 
+strings. All functions can also accept a throwable before other normal 
+arguments to log a specific exception.
+
+
+## Testing
+clojure.tools.logging provides a [test namespace](http://clojure.github.io/tools.logging/#clojure.tools.logging.test).
+
+An example of usage is in `test/zero/unit/logging_test.clj`. 
+That file doesn't actually test our code, but rather our dependencies:
+it helps catch issues that might arise from clojure.tools.logging (or SLF4J, which
+it currently delegates to) having trouble finding a logging implementation via
+service loaders and the like. It isn't bulletproof--it is theoretically possible
+for tests to use a different classpath than other execution--but it is a good smoke
+test that in that any failures there would require remediation in our own codebase.
+
+## Under the Hood
+WFL's logging works as follows:
```

Comment:
This a bit verbose and repetitive. I wouldn't go into so much detail and try to avoid repeating yourself.

Reply:
Added a summary at the top of the document.

I think I might be unique in terms of my want of documentation in WFL. I've found WFL to be extremely hard to get up to speed with, and I think it is a shame that the most useful resource I have found for understanding it has been looking at Tom's complaint tickets in the backlog--he mentions things that are superfluous and that begins to knock on the "why" of how on earth this software is built.

I'm happy to remove what I've written here and put it on my desktop instead--I'm going to maintain it regardless.

Reply:
Yeah wfl does have a number of things that make me scratch my head too! There are a number of things I don't understand yet.
Feel free to document, by all means. I think stuff like this is really useful but it could be a bit more concise. If you're exposing `clojure.tools.logging`, I would state we use that and its configuration with a brief explanation for those choices.

Reply:
(Updated the document to make it more brief after a conversation with Ed last week)